corehq/apps/export/views/utils.py

```diff
@@ -130,9 +130,14 @@ def has_deid_view_permissions(self):
         # just a convenience wrapper around user_can_view_deid_exports
         return user_can_view_deid_exports(self.domain, self.couch_user)
 
-    def access_list_exports_or_404(self, is_deid=False):
+    @property
+    def has_odata_permissions(self):
+        return user_can_view_odata_feed(self.domain, self.couch_user)
+
+    def access_list_exports_or_404(self, is_deid=False, is_odata=False):
```

Comment:
nit: I think this might be a bit clearer to move `is_deid` and `is_odata` into the class constructor.  The `permissions` object is already specific to a `model_type` - seems like it should have all the context up front.

Reply:
not sure i follow...this information isn't necessarily present at the point where `ExportsPermissionsManager` is called. Also, sometimes this information is in the `GET` or `POST` body...so where it comes from is not necessarily consistent

Reply:
I meant to pass it in already parsed from the request however appropriate for the view.  From the examples I looked at it seemed like this was pretty much always initialized right before it was used (ie, when the request was already in scope), though I didn't look at all usages, so maybe it doesn't actually work like that.

Reply:
yeah it the view hierarchy it starts getting a little weird for that...